### ISSUE TITLE
Issue with loss_function='MAPE'

### ISSUE THREAD

**Issue by david-waterworth** [catboost/catboost#593]
Problem: 

The following estimator seems to correctly minimise the test set MAPE.

```
model = CatBoostRegressor(n_estimators=1000, max_depth=3, loss_function='RMSE',eval_metric='MAPE', has_time=True, logging_level='Verbose', use_best_model=True,metric_period=100)
model.fit(train, eval_set=valid, plot=False)
```
> 0:	learn: 0.9689602	test: 0.9694422	best: 0.9694422 (0)	total: 16.6ms	remaining: 16.6s
> 100:	learn: 0.0776661	test: 0.0816106	best: 0.0816106 (100)	total: 1.51s	remaining: 13.5s
> 200:	learn: 0.0696934	test: 0.0701527	best: 0.0701527 (200)	total: 2.82s	remaining: 11.2s
> ...
> 800:	learn: 0.0632865	test: 0.0659328	best: 0.0659328 (800)	total: 9.53s	remaining: 2.37s
> 900:	learn: 0.0630847	test: 0.0658580	best: 0.0658580 (900)	total: 10.6s	remaining: 1.17s
> 999:	learn: 0.0629461	test: 0.0657494	best: 0.0657494 (999)	total: 11.7s	remaining: 0us
> 
> bestTest = 0.06574943909
> bestIteration = 999

However when I change loss_function from 'RMSE' to 'MAPE' it does seems to progress incredibly slowly. 

> 0:	learn: 0.9999891	test: 0.9999892	best: 0.9999892 (0)	total: 16.6ms	remaining: 13m 49s
> 100:	learn: 0.9989047	test: 0.9989085	best: 0.9989085 (100)	total: 1.45s	remaining: 11m 58s
> 200:	learn: 0.9978209	test: 0.9978304	best: 0.9978304 (200)	total: 2.9s	remaining: 11m 58s
> 300:	learn: 0.9967363	test: 0.9967494	best: 0.9967494 (300)	total: 4.33s	remaining: 11m 55s
> 400:	learn: 0.9956494	test: 0.9956671	best: 0.9956671 (400)	total: 5.77s	remaining: 11m 54s
> 500:	learn: 0.9945570	test: 0.9945773	best: 0.9945773 (500)	total: 7.2s	remaining: 11m 51s
> 600:	learn: 0.9934638	test: 0.9934874	best: 0.9934874 (600)	total: 8.61s	remaining: 11m 47s
> 700:	learn: 0.9923707	test: 0.9923981	best: 0.9923981 (700)	total: 10s	remaining: 11m 43s
> 800:	learn: 0.9912778	test: 0.9913098	best: 0.9913098 (800)	total: 11.4s	remaining: 11m 39s
> 900:	learn: 0.9901848	test: 0.9902208	best: 0.9902208 (900)	total: 12.8s	remaining: 11m 36s
> 1000:	learn: 0.9890918	test: 0.9891322	best: 0.9891322 (1000)	total: 14.2s	remaining: 11m 33s

Increasing learning_rate to 10 resulted in decent convergence but there's a warning raised (learning_rate > 1). Is this expected?

> 0:	learn: 0.9963755	test: 0.9964094	best: 0.9964094 (0)	total: 16.8ms	remaining: 13m 59s
> 100:	learn: 0.6320917	test: 0.6335425	best: 0.6335425 (100)	total: 1.35s	remaining: 11m 8s
> 200:	learn: 0.3002645	test: 0.3038692	best: 0.3038692 (200)	total: 2.7s	remaining: 11m 8s
> 300:	learn: 0.1610344	test: 0.1674332	best: 0.1674332 (300)	total: 4.08s	remaining: 11m 13s
> 400:	learn: 0.0922991	test: 0.0977295	best: 0.0977295 (400)	total: 5.34s	remaining: 11m
> 500:	learn: 0.0707975	test: 0.0770296	best: 0.0770296 (500)	total: 6.67s	remaining: 10m 59s
> 600:	learn: 0.0632390	test: 0.0702062	best: 0.0702062 (600)	total: 8.01s	remaining: 10m 58s
> 700:	learn: 0.0583728	test: 0.0659457	best: 0.0659457 (700)	total: 9.33s	remaining: 10m 56s
> 800:	learn: 0.0562854	test: 0.0645781	best: 0.0645781 (800)	total: 10.7s	remaining: 10m 54s
> 900:	learn: 0.0540739	test: 0.0631779	best: 0.0631779 (900)	total: 12s	remaining: 10m 56s
> 1000:	learn: 0.0526777	test: 0.0628055	best: 0.0628055 (1000)	total: 13.4s	remaining: 10m 57s
> 

catboost version: 0.11.2
Operating System: Ubuntu 18.04

**Comment by annaveronika** [comment]
I'm looking through old issues, very sorry for answering so long! Could you please share your dataset, so that we could look on it?

**Comment by david-waterworth** [comment]
Hi @annaveronika - This has been fixed by `leaf_estimation_method=Exact`, I don't have the original data set this occurs for most of my models using Newton boosting (setting a learning rate > 1 helped). Using `leaf_estimation_method=Exact' and `task_type='CPU` produces a huge improvement.
